.pre-commit-config.yaml

```diff
@@ -0,0 +1,49 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+repos:
```

Comment:
Since our org-wise Python linting setup isn't really set in stone yet (and because this is a special-case repo), I don't think it's super important for now that it conforms to the style of our other primary user-facing Python tools (i.e., the [api](https://github.com/neurobagel/api), [cli](https://github.com/neurobagel/bagel-cli), and [dashboard](https://github.com/neurobagel/proc_dash)). That said, here is the [pre-commit config](https://github.com/neurobagel/api/blob/main/.pre-commit-config.yaml) that we're currently using across the other Python tools, if you want to have a look.

I think a minimal change (if you want) could be to update the max line length for flake8/black to be consistent with our config, since I think this will typically introduce the most diffs if we decide to align it later on.

Most likely, any additions/changes we do make following https://github.com/neurobagel/project/issues/107 will use our existing pre-commit config as a template. 🙂

Reply:
will update the line length to match that of other repos and try to match the config of the hooks that are common 

some of the extra checks I don't think interfere much with actual style but are more there to ensure that files are parseable and python code does not rely on "antiquated" python syntax.

in terms of set up, I actually think that it would be better to have use dedicated config files (pyproject.toml, .flake8...) for each linting tool otherwise you may get different results if you run them via pre-commit or not.

see https://github.com/neurobagel/project/issues/114

Reply:
Agreed! When we revisit that linting issue we can use this repo as a guideline as well